### PR TITLE
Feature/use p button instead of alg button

### LINKED ISSUE #961: Use p-button instead of alg-button
## Overview

I haven't found used secondary and warning buttons.

## Acceptance criteria
- [ ]

1. Clarify buttons scss file, remove unused code

### PR DESCRIPTION
**Pull request by Iloveall**
## Description

Fixes #961

## Notes (out of scope, known isues, hints for reviewing code, ...)  (optional)

...

## Test cases

- [x] Case 1:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/use-p-button-instead-of-alg-button/en/#/activities/by-id/1980584647557587953;path=4702,4102;attempId=0/edit)
  3. Change description field
  4. Then I see "Save" and "Cancel Changes" buttons in bottom

- [x] Case 2:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/use-p-button-instead-of-alg-button/en/#/groups/by-id/50118905546577977;path=/details/members)
  3. Scroll down to "Invite Users" section
  4. Then I see "Invite" button

- [x] Case 3:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/use-p-button-instead-of-alg-button/en/#/groups/by-id/672913018859223173;path=52767158366271444/details)
  3. Scroll down to "Leave the group?" section
  4. Then I see "Leave This Group" button

- [x] Case 4:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/use-p-button-instead-of-alg-button/en/#/groups/mine)
  3. Scroll down to "Join a group by code" section
  4. Then I see "Join" button with icon

- [x] Case 5:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/use-p-button-instead-of-alg-button/en/#/groups/by-id/4035378957038759250;path=/details/managers)
  3. Scroll down to "Add a manager" section
  4. Then I see "Add" button with icon

- [x] Case 6:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/use-p-button-instead-of-alg-button/en/#/groups/by-id/672913018859223173;path=52767158366271444/details)
  3. Then I start watch group
  4. Then I click on "test 2 (Pixal)" of suggestions
  5. And I click on "«Pixal» can access this activity" header
  6. Then I click on "Edit Permissions" button
  7.  Then I see "Cancel" and "Proceed" buttons with icons

- [x] Case 7:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/use-p-button-instead-of-alg-button/en/#/groups/by-id/672913018859223173;path=52767158366271444/details/managers)
  3. Then I click on edit on one of the manager in managers list
  4.  Then I see "Cancel" and "Save" buttons with icons

- [x] Case 8:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/use-p-button-instead-of-alg-button/en/#/activities/by-id/1980584647557587953;path=4702,4102;attempId=0/edit)
  3. Then I scroll down to the "Add a content" section
  4. Then I search by word "Additions"
  5. Then I see "Add" button with icon

- [x] Case 9:
  1. Given I am the usual user
  2. When I go to [this page](https://dev.algorea.org/branch/feature/use-p-button-instead-of-alg-button/en/#/groups/by-id/2116767266653043949;path=/details/members)
  3. Then I scroll down to the "Let user join using a code" section
  4. Then I see "Generate a code" button with icon

### CODE REVIEW

File: src/app/modules/shared-components/components/access-code-view/access-code-view.component.html [hunk 1]
```diff
@@ -11,12 +11,12 @@
...
+    <p-button
+      styleClass="alg-button p-button-rounded"
       icon="fa fa-check"
       [label]="buttonLabel"
-      class="p-button-rounded"
       [disabled]="input.value === '' || state === 'loading'"
-      (click)="onClickAccess()"
-    ></alg-button>
+      (onClick)="onClickAccess()"
+    ></p-button>
```
Comment (ThomasJuster) [inline src/app/modules/shared-components/components/access-code-view/access-code-view.component.html:20]: The don't look the same, is that intended?
Before:
image

After:
image

File: src/app/modules/group/components/group-join-by-code/group-join-by-code.component.html [hunk 2]
```diff
@@ -23,14 +23,14 @@
       <span class="label red" *ngIf="!codeInfo.hasCodeNotSet && codeInfo.hasCodeExpired" i18n>
         This code has expired on {{ codeInfo.codeExpiration | date:'medium' }}
       </span>
-      <alg-button
-        label="Generate a code"
+      <p-button
```
Comment (ThomasJuster) [inline src/app/modules/group/components/group-join-by-code/group-join-by-code.component.html:26]: I didn't see a test case for this file :confused:

File: src/app/modules/group/components/group-join-by-code/group-join-by-code.component.html [hunk 3]
```diff
@@ -61,15 +61,15 @@
...
-            class="submit-duration-button"
-            (click)="codeLifetimeControlValue && submitCodeLifetime(codeLifetimeControlValue.ms)"
+            styleClass="alg-button submit-duration-button"
+            (onClick)="codeLifetimeControlValue && submitCodeLifetime(codeLifetimeControlValue.ms)"
             [disabled]="processing || !codeLifetimeControlValue || !control.dirty || codeLifetimeControlValue.ms === group.codeLifetime?.ms"
             [pTooltip]="durationTooltip"
             tooltipPosition="right"
             tooltipEvent="hover"
-          ></alg-button>
+          ></p-button>
```
Comment (ThomasJuster) [inline src/app/modules/group/components/group-join-by-code/group-join-by-code.component.html:72]: url: https://dev.algorea.org/branch/feature/use-p-button-instead-of-alg-button/en/#/groups/by-id/672913018859223173;path=52767158366271444/details/members
The last button on the right is too high and too wide compared to before
Before:
image

After:
image

File: src/app/modules/group/components/group-join-by-code/group-join-by-code.component.html [hunk 4]
```diff
@@ -23,14 +23,14 @@
...
+        i18n-label label="Generate a code"
         icon="fa fa-plus"
-        class="p-button-rounded"
-        (click)="generateNewCode()"
+        styleClass="alg-button p-button-rounded"
+        (onClick)="generateNewCode()"
         [disabled]="processing"
         *ngIf="codeInfo.hasCodeNotSet"
-      ></alg-button>
+      ></p-button>
```
Comment (ThomasJuster) [inline src/app/modules/group/components/group-join-by-code/group-join-by-code.component.html:33]: Is the style change intended?
Before:
image

After:
image

**Review by ThomasJuster — COMMENTED** [review]
I'll stop the review here because I'll always ask if the button style change is intended and that's not very productive.
Nice code refactoring :+1:

File: src/app/modules/group/components/group-join-by-code/group-join-by-code.component.html (hunk 2 above)
Comment (Iloveall) [reply src/app/modules/group/components/group-join-by-code/group-join-by-code.component.html:26]: I've add one more test case

File: src/app/modules/group/components/group-join-by-code/group-join-by-code.component.html (hunk 3 above)
Comment (Iloveall) [reply src/app/modules/group/components/group-join-by-code/group-join-by-code.component.html:72]: Fixed height for button

File: src/app/modules/group/components/group-leave/group-leave.component.html [hunk 5]
```diff
@@ -11,12 +11,12 @@
...
-      class="p-button-rounded-full"
+    <p-button
+      styleClass="alg-button p-button-rounded-full"
       icon="fa fa-times"
       i18n-label label="Leave this group"
-      (click)="leaveGroup()"
+      (onClick)="leaveGroup()"
       [disabled]="group.isMembershipLocked"
-    ></alg-button>
+    ></p-button>
```
Comment (ThomasJuster) [inline src/app/modules/group/components/group-leave/group-leave.component.html:20]: The button icon alignment looks broken, can you check if an issue exists for that and if not create one?
image
I spotted the misalignment in other buttons, so I guess it's at the button level:
Manager add button: 
image

File: src/app/modules/shared-components/components/add-content/add-content.component.html [hunk 6]
```diff
@@ -55,13 +55,13 @@ <h3 i18n>Select a Content</h3>
...
-              [icon]="addedIds.includes(item.id) ? 'fa fa-check' : selectExistingText === 'Add' ? 'fa fa-plus' : undefined"
+            <p-button
+              styleClass="alg-button p-small-button p-button-rounded"
+              [icon]="addedIds.includes(item.id) ? 'fa fa-check' : selectExistingText === 'Add' ? 'fa fa-plus' : ''"
               [label]="addedIds.includes(item.id) ? addedText : selectExistingText"
               [disabled]="addedIds.includes(item.id)"
-              (click)="addExisting(item)"
-            ></alg-button>
+              (onClick)="addExisting(item)"
+            ></p-button>
```
Comment (ThomasJuster) [inline src/app/modules/shared-components/components/add-content/add-content.component.html:64]: Vertical alignment of buttons "add" and "already added" is odd, can you check if there's an issue for that and create one if there's none?
image

File: src/app/modules/item/pages/item-display/item-display.component.html [hunk 7]
```diff
@@ -28,16 +28,18 @@
...
         i18n-label label="Show task anyway (for debugging)"
         (click)="showTaskAnyway = true"
-      ></alg-button>
-      <alg-button
+      ></p-button>
+      <p-button
         class="error-button"
+        styleClass="alg-button"
         i18n-label label="Retry"
         (click)="refresh.emit()"
```
Comment (ThomasJuster) [inline src/app/modules/item/pages/item-display/item-display.component.html:41]: I'm confused, everywhere else you changed `(click)` to (onClick)` but not here and it still works, so both `(click)` and `(onClick)` are ok to use with p-button?

**Review by ThomasJuster — APPROVED** [review]

File: src/app/modules/item/pages/item-display/item-display.component.html (hunk 7 above)
Comment (Iloveall) [reply src/app/modules/item/pages/item-display/item-display.component.html:41]: (click) doesn't block event when disabled state

**Pull request merged**